pkgs/build-support/node/build-npm-package/hooks/npm-config-hook.sh

```diff
@@ -0,0 +1,76 @@
+npmSetupHook() {
+    echo "Executing npmSetupHook"
+
+    runHook preConfigure
+
+    echo "Configuring npm"
+
+    export HOME=$TMPDIR
+    export npm_config_nodedir="@nodeSrc@"
+
+    npm config set cache "$npmDeps"
+    npm config set offline true
+    npm config set progress false
+
+    local cacheLockfile="$npmDeps/package-lock.json"
+    local srcLockfile="$PWD/package-lock.json"
+
+    echo "Validating consistency between $srcLockfile and $cacheLockfile"
+
+    if ! @diff@ "$srcLockfile" "$cacheLockfile"; then
+      # If the diff failed, first double-check that the file exists, so we can
+      # give a friendlier error msg.
+      if ! [ -e "$srcLockfile" ]; then
+        echo "ERROR: Missing package-lock.json from src. Expected to find it at: $srcLockfile"
+        echo "Hint: You can use the patches attribute to add a package-lock.json manually to the build."
+        exit 1
+      fi
+
+      if ! [ -e "$cacheLockfile" ]; then
+        echo "ERROR: Missing lockfile from cache. Expected to find it at: $cacheLockfile"
+        exit 1
+      fi
+
+      echo
+      echo "ERROR: npmDepsHash is out of date"
+      echo
+      echo "The package-lock.json in src is not the same as the in $npmDeps."
+      echo
+      echo "To fix the issue:"
+      echo '1. Use `lib.fakeHash` as the npmDepsHash value'
+      echo "2. Build the derivation and wait for it to fail with a hash mismatch"
+      echo "3. Copy the 'got: sha256-' value back into the npmDepsHash field"
+      echo
+
+      exit 1
+    fi
+
+    echo "Installing dependencies"
+
+    npm ci --ignore-scripts
```

Comment:
Might be useful to add a comment explaining why we ignore scripts here but run them manually later.